CMakeLists.txt

```diff
@@ -333,11 +334,12 @@ int main() {
   auto d = _mm_cvtsi128_si64(c);
 }
 " HAVE_SSE42)
-if(HAVE_SSE42)
-  add_definitions(-DHAVE_SSE42)
-  add_definitions(-DHAVE_PCLMUL)
-elseif(FORCE_SSE42)
-  message(FATAL_ERROR "FORCE_SSE42=ON but unable to compile with SSE4.2 enabled")
+  if(HAVE_SSE42)
+    add_definitions(-DHAVE_SSE42)
+    add_definitions(-DHAVE_PCLMUL)
```

Comment:
(Pre-existing issue) According to this website I found, Nehalem processors have SSE 4.2 but not PCLMUL

https://en.wikipedia.org/wiki/Nehalem_(microarchitecture)

Reply:
Yeah, there is an issue on this topic still open https://github.com/facebook/rocksdb/issues/3782#issuecomment-386445675. We knew about the possibility of such HW when introducing 3-way crc32c too but chose not to handle it then...